epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java

```diff
@@ -603,6 +604,44 @@ private MethodSpec generateSetter(ClassToGenerateInfo helperClass, AttributeInfo
         .build();
   }
 
+  private MethodSpec generateReset(ClassToGenerateInfo helperClass) {
+    Builder builder = MethodSpec.methodBuilder("reset")
+        .addAnnotation(Override.class)
+        .addModifiers(Modifier.PUBLIC)
+        .returns(helperClass.getParameterizedGeneratedName())
+        .addStatement("super.reset()");
+
+    for (AttributeInfo attributeInfo : helperClass.getAttributeInfo()) {
+      if (!attributeInfo.hasFinalModifier()) {
+        builder.addStatement("this.$L = $L", attributeInfo.getName(),
+            getDefaultValue(attributeInfo.getType()));
+      }
+    }
+
+    return builder
+        .addStatement("return this")
+        .build();
+  }
+
+  private String getDefaultValue(TypeName attributeType) {
+    String defaultValue;
```

Comment:
no need for the `String defaultValue`, intermediate variable. could just return directly in each if block